src/main/java/step1/Splitter.java

```diff
@@ -0,0 +1,39 @@
+package step1;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Splitter {
+
+    private static final String COMMA_DELIMITER = ",";
+    private static final String COLON_DELIMITER = ":";
+    private static final String DEFAULT_DELIMITER_REGEX = ",|:";
+    private static final String CUSTOM_DELIMITER_REGEX = "//(.)\n(.*)";
+
+    public String[] split(String text) {
+        return getSplitText(text);
+    }
+
+    private String[] getSplitText(String text) {
+        if(text.contains(COLON_DELIMITER) || text.contains(COMMA_DELIMITER)) {
+            return getDefaultSplitText(text);
+        }
+        if(text.matches(CUSTOM_DELIMITER_REGEX)) {
+            return getCustomSplitText(text);
+        }
+        throw new RuntimeException(Error.DELIMITER_NOT_EXIST);
+    }
+
+    public String[] getDefaultSplitText(String text) {
+        return text.split(DEFAULT_DELIMITER_REGEX);
+    }
+
+    public String[] getCustomSplitText(String text) {
+        Matcher matcher = Pattern.compile(CUSTOM_DELIMITER_REGEX).matcher(text);
```

Comment:
Pattern은 비싼 객체(객체를 만드는데 비용이 많이 듬)기 때문에, 아래와 같이 상수로 정의해서 사용하는 건 어떨까요?

```suggestion
    private static final Pattern SPLIT_KEY_PATTERN = Pattern.compile("//(.)\n(.*)");
```

```java
public class RomanNumber {

    private static final Pattern ROMAN = Pattern.compile("^(?=.)M*(C[MD]|D?C{0,3})(X[CL]|L?X{0,3})(I[XV]|V?I{0,3})$");

    static boolean isRomanNumeral(String s) {
        return ROMAN.matcher(s).matches();
    }
}
```

링크 : https://github.com/keesun/study/blob/master/effective-java/item6.md

Reply:
엊그제 이펙티브 자바에서 봤던 내용이었는데 미처 생각하지 못했네요. static으로 한번만 만들어 놓으면 계속 사용할 수 있게 되네요. 다음에 비슷한 상황에 사용해봐야 될 것 같습니다👍